chore(editor): Validate SSO metadata url

afitzek opened the pull request:
## Summary

Validate the SSO metadata url before saving it on the server

## Related Linear tickets, Github issues, and Community forum posts

fixes [PAY-2490](https://linear.app/n8n/issue/PAY-2490)

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [X] Tests included.
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

afitzek committed:
Add input validation for saml metadata url

We want to validate the provided input of the user when they are setting
up SSO integration with SAML. This validates that the provided input
is a valid URL with a valid protocol (http, or https).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/n8n-io/n8n?pullRequest=15206) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/n8n-io/n8n?pullRequest=15206) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/n8n-io/n8n?pullRequest=15206) it.</sub>

afitzek commented on packages/frontend/editor-ui/src/views/SettingsSso.test.ts:
```diff
@@ -194,6 +194,64 @@ describe('SettingsSso View', () => {
+		const urlinput = getByTestId('sso-provider-url');
+
+		expect(urlinput).toBeVisible();
+		await userEvent.type(urlinput, samlConfig.metadata!);
```
This is by design, the reason for this test is to make sure that if a string that is not an url is provided as input, an error is shown and the settings are not saved onto the server.

MarcL reviewed (commented):
A few minor comments.

afitzek committed:
Adapt texts and test names

Explain that we want to allow http for the saml metadata url for now as well,
to avoid a theoretical breaking change.
